synapse/config/server.py

```diff
@@ -184,49 +184,74 @@ def generate_ip_set(
 
 @attr.s(frozen=True)
```

Comment:
This doesn't use `slots` since we splat a dictionary directly into it from the user (e.g. `HttpResourceConfig(**config)`).